src/sync/app.cpp

```diff
@@ -36,7 +36,7 @@ Optional<T> get_optional(const nlohmann::json& json, const std::string& key)
     return it != json.end() ? Optional<T>(it->get<T>()) : realm::util::none;
 }
 
-static std::map<std::string, std::string> get_request_headers(std::shared_ptr<SyncUser> with_user_authorization = nullptr)
+static std::map<std::string, std::string> get_request_headers(std::shared_ptr<SyncUser> with_user_authorization = nullptr, bool use_access_token = false)
```

Comment:
These settings are a bit confusing to read at this point, since it in reality is 3 different modes. Perhaps extract it into a single enum `AuthMode` with 3 settings : `NoAuth`, `RefreshToken` `AccessToken`? This was just the first names that popped into my head, so feel free to change.

Reply:
I've added an enum